src/main/java/io/stargate/sgv2/jsonapi/exception/playing/APIException.java

```diff
@@ -0,0 +1,128 @@
+package io.stargate.sgv2.jsonapi.exception.playing;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+/**
+ * Base for any exceptions from the API.
+ *
+ * <p>All errors are of a {@link ErrorFamily}, this class should not be used directly, one of the
+ * subclasses should be used. There are further categorised to be errors have an optional {@link
+ * ErrorScope}, that groups errors of a similar source together. Finally, the error has an {@link
+ * ErrorCode} that is unique within the family and scope.
+ *
+ * <p>To facilitate better error messages we template the messages in a {@link ErrorTemplate} that
+ * is loaded from a properties file. The body for the error may change with each instance of the
+ * exception, for example to include the number of filters that were included in a request.
+ *
+ * <p>The process of adding a new Error Code is:
+ *
+ * <p>
+ *
+ * <ul>
+ *   <li>Decide what {@link ErrorFamily} the code belongs to.
+ *   <li>Decide if the error has a {@link ErrorScope}, such as errors with Embedding Providers, if
+ *       it does not then use {@link ErrorScope#NONE}.
+ *   <li>Decide on the {@link ErrorCode}, it should be unique within the Family and Scope
+ *       combination.
+ *   <li>Add the error to file read by {@link ErrorTemplate} to define the title and templated body
+ *       body.
+ *   <li>Add the error code to the Code enum for the Exception class, such as {@link
+ *       FilterException.Code} or {@link RequestException.Code} with the same name. When the enum is
+ *       instantiated at JVM start the error template is loaded.
+ *   <li>Create instances using the methods on {@link ErrorCode}.
+ * </ul>
+ *
+ * To get the Error to be returned in the {@link
+ * io.stargate.sgv2.jsonapi.api.model.command.CommandResult} call the {@link #get()} method to get a
+ * {@link CommandResponseError} that contains the subset of information we want to return.
+ */
+public abstract class APIException extends RuntimeException
+    implements Supplier<CommandResponseError> {
+
+  // All errors default to 200 HTTP status code, because we have partial failure modes.
+  // There are some overrides, e.g. a server timeout may be a 500, this is managed in the
+  // error config. See ErrorTemplate.
+  public static final int DEFAULT_HTTP_RESPONSE = 200;
+
+  /**
+   * HTTP Response code for this error. NOTE: Not using enum from quarkus because do not want
+   * references to the HTTP framework this deep into the command processing
+   */
+  public final int httpResponse;
```

Comment:
I think it may make sense to keep as `int` to avoid conversion on deserialization from YAML.

But FWTW, it wouldn't be (or have to be) Quarkus dep: standard Java JAX-RS has `jakarta.ws.rs.core.Response.Status` that is commonly used (Quarkus is a JAX-RS / Jakarta-RS implementation) as enum for HTTP status code.